src/rgw/driver/rados/config/zone.cc

```diff
@@ -163,31 +163,90 @@ class RadosZoneWriter : public sal::ZoneWriter {
     return 0;
   }
 }; // RadosZoneWriter
+ 
 
+int RadosConfigStore::get_zones_pool_set(const DoutPrefixProvider *dpp,
+                       optional_yield y,
+                       std::string_view my_zone_id,
+                       std::set<rgw_pool>& pools)
+{
+  std::array<std::string, 128> zone_names;
+  rgw::sal::ListResult<std::string> listing;
+  do {
+    int r = list_zone_names(dpp, y, listing.next,
+                                      zone_names, listing);
+    if (r < 0) {
+      ldpp_dout(dpp, 0) << "failed to list zones with " << cpp_strerror(r) << dendl;
+      return r;
+    }
+
+    for (const auto& name : listing.entries) {
+      RGWZoneParams info;
+      r = read_zone_by_name(dpp, y, name, info, nullptr);
+      if (r < 0) {
+        ldpp_dout(dpp, 0) << "failed to load zone " << name
+            << " with " << cpp_strerror(r) << dendl;
+        return r;
+      }
+      if (info.get_id() != my_zone_id) {
+        rgw::add_zone_pools(info, pools);
+      }
+    }
+  } while (!listing.next.empty());
+
+  return 0;
+}
+
+// TODO: remove this and add it to namespace rgw
+static std::string gen_random_uuid()
+{
+  uuid_d uuid;
+  uuid.generate_random();
+  return uuid.to_string();
+}
 
 int RadosConfigStore::create_zone(const DoutPrefixProvider* dpp,
```

Comment:
@dang Is this where you envisioned the default sal config (from https://github.com/ceph/ceph/pull/52665) being added to the RGWZoneParams sal_config?